docs/learn/programming-guide.md

```diff
@@ -0,0 +1,414 @@
+---
+layout: default
+---
+<p>
+  <div class="alert alert-info alert-dismissible" role="alert">
+  <span class="glyphicon glyphicon-flag" aria-hidden="true"></span>
+  <button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+  The Apache Beam project is in the process of bootstrapping. This includes the creation of project resources, the refactoring of the initial code submission, and the formulation of project documentation, planning, and design documents. For more information about Beam see the <a href="/getting_started/">getting started page</a>.
+  </div>
+</p>
+
+# Beam Programming Guide
+
+The **Beam Programming Guide** is intended for Beam users who want to use the Beam SDKs to create data processing pipelines. It contains guidance for using the Beam SDK classes to build and test your pipeline. It is not intended as an exhaustive reference, but as a higher-level guide to programmatically building your Beam pipeline.
```

Comment:
I'd vote for a more explicit disclaimer that this is a language-independent guide. (And at some point a mention of toggling code snippets?) May also want to point to the SDKs landing page for things like JavaDoc (once we create it ;)